.circleci/config.yml

```diff
@@ -57,6 +57,9 @@ jobs:
           name: "Ensure latest docs build without warnings or errors"
           command: >-
             poetry install -E docs &&
+            poetry run pip install ipython &&
```

Comment:
@kmetallic04 can we move this to pyproject.toml so it is installed with the command above?

Reply:
tested above, does not work with poetry. this should be good to go if this is the only method to install python on ci

Reply:
Hmmm.... I think there's a version of `ipython` (version `7.16.1`) that's compatible with the current requirements for python within poetry (`>=3.6.1,<4.0.0`). I think we can update `pyproject.toml` and `poetry.lock` instead.

Reply:
Done. Added `ipython` as an `extra` for `recipes`. No need for using `pip`.